addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/InputWithNavigationControl.java

```diff
@@ -0,0 +1,439 @@
+        for (int i = 0; i < MAX_PER_PAGE; i++) {
+            if (itemName.equals(cache.items[i])) {
+                return i + 1;
+            }
+        }
+
+        return -1;
+    }
+
+    private boolean selectItem(String name) throws IOException, ParserConfigurationException, SAXException {
+        final int pageCount = (int) Math.ceil(cache.Max_Line / (double) MAX_PER_PAGE);
+        final int currentPage = (int) Math.floor((cache.Current_Line - 1) / (double) MAX_PER_PAGE);
+
+        HttpXMLSendReceive com = com_ref.get();
+        for (int pageIndex = 0; pageIndex < pageCount; pageIndex++) {
+            // Start with the current page and then go to the end and start at page 1 again
+            int realPage = (currentPage + pageIndex) % pageCount;
+
+            if (currentPage != realPage) {
+                goToPage(pageIndex);
+            }
+
+            int index = findItemOnCurrentPage(name);
+            if (index > 0) {
+                com.postPut(wrInput(
+                        "<List_Control><Direct_Sel>Line_" + String.valueOf(index) + "</Direct_Sel></List_Control>"));
+                updateNavigationState();
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Refreshes the menu state and caches the List_Info node from the response. This method may take
+     * some time because it retries the request for up to MENU_MAX_WAITING_TIME or the menu state reports
+     * "Ready", whatever comes first.
+     *
+     * @throws Exception
+     */
+    public void updateNavigationState() throws IOException, ParserConfigurationException, SAXException {
+        int totalWaitingTime = 0;
+
+        Document doc;
+        Node currentMenu;
+
+        HttpXMLSendReceive com = com_ref.get();
+        while (true) {
+            String response = com.post(wrInput("<List_Info>GetParam</List_Info>"));
+            doc = com.xml(response);
+            if (doc.getFirstChild() == null) {
+                throw new SAXException("<Play_Control>GetParam failed: " + response);
+            }
+
+            currentMenu = HttpXMLSendReceive.getNode(doc.getFirstChild(), "List_Info");
+
+            if (currentMenu == null) {
+                throw new SAXException("<List_Info>: GetParam response invalid!");
+            }
+
+            Node nodeMenuState = HttpXMLSendReceive.getNode(currentMenu, "Menu_Status");
+
+            if (nodeMenuState == null || nodeMenuState.getTextContent().equals("Ready")) {
+                break;
+            }
+
+            totalWaitingTime += ZoneControl.MENU_RETRY_DELAY;
+            if (totalWaitingTime > ZoneControl.MENU_MAX_WAITING_TIME) {
+                throw new IOException("Menu still not ready after " + ZoneControl.MENU_MAX_WAITING_TIME + "ms");
+            }
+
+            try {
+                Thread.sleep(ZoneControl.MENU_RETRY_DELAY);
+            } catch (InterruptedException e) {
+                // Ignore and just retry immediately
```

Comment:
If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/